shub/image/build.py

```diff
@@ -61,7 +64,12 @@ def build_cmd(target, version, skip_tests):
     else:
         build_progress_cls = _BuildProgress
     click.echo("Building {}.".format(image_name))
-    events = client.build(path=project_dir, tag=image_name, decode=True)
+    events = client.build(
+        path=project_dir,
+        tag=image_name,
+        decode=True,
+        dockerfile=filename
```

Comment:
Should this be `dockerfile=dockerfile`? I guess `client.build` will default to `Dockerfile` anyways, but still, I would rather not use two different variables for the same thing.

Reply:
Or is there some "smart Dockerfile finding" that we would miss out on when supplying `'Dockerfile'` instead of `None`?

Reply:
I'm not 100% sure about the logic applied by Docker, but I was using None instead of 'Dockerfile' because None is the default value in docker-py's build method: https://github.com/docker/docker-py/blob/master/docker/api/build.py#L16

Let me know if you prefer to change that.

Reply:
Hm, that line is over two years old (a very long time before we even introduced custom images for Scrapy Cloud), and the error message would be quite cryptic to `shub` users anyways. I'd prefer to enforce `docker-py>=1.17` in our `setup.py` and `requirements.in` instead.

I realized we're not using any smart Dockerfile-finding by `client.build()` anyways because we explicitly enforce the existence of "project_dir/Dockerfile" a few lines above. Given all this I'd prefer to set defaults in the click option and the function parameter list, and avoid the dual variables.